src/lib/tabs/tab-header.ts

```diff
@@ -0,0 +1,258 @@
+import {
+  ViewChild,
+  Component,
+  Input,
+  NgZone,
+  QueryList,
+  ElementRef,
+  Renderer, ViewEncapsulation, ContentChildren, Output, EventEmitter
+} from '@angular/core';
+import {RIGHT_ARROW, LEFT_ARROW, ENTER, Dir, LayoutDirection} from '../core';
+import {MdTabLabelWrapper} from './tab-label-wrapper';
+import {MdInkBar} from './ink-bar';
+import 'rxjs/add/operator/map';
+
+export type ScrollDirection = 'after' | 'before';
+
+@Component({
+  moduleId: module.id,
+  selector: 'md-tab-header',
+  templateUrl: 'tab-header.html',
+  styleUrls: ['tab-header.css'],
+  encapsulation: ViewEncapsulation.None,
+  host: {
+    '[class.md-tab-header]': 'true',
+    '[class.scroll]': '_isScrollingEnabled()',
```

Comment:
Agreed, I think init and tabs add/remove should handle the vast majority of use cases. I'll include just that logic in this PR.

In another PR, I'll attempt to check tab heading changes (text node modifications) using the `MutationObserver`.

I think one more thing to consider is that we could check when a change is made to the selected tab or focused tab. This would provide a fallback that could be triggered by the user who may try to get the header to update the scrolling after some event has caused the width to change.